src/Control/Monad/Catch.hs

```diff
@@ -474,8 +474,8 @@ instance (Error e, MonadMask m) => MonadMask (ErrorT e m) where
           Right resource -> runErrorT (release resource) >> return ())
       (\eresource e ->
          case eresource of
-           Left _ -> throwM e
-           Right resource -> runErrorT (cleanup resource e))
+           Left _ -> return ()
```

Comment:
To make sure I understand this change (from `throwM e` to `return ()`) properly, it's because the exception `e` will be rethrown regardless?

Reply:
Yes, correct.